World.py

```diff
@@ -187,6 +191,158 @@ def __init__(self, id, settings):
+                        gbk.goals.append(Goal(self, { 'replace': 'Spirit Medallion' }, 'Yellow', items=[{'name': 'Spirit Medallion','quantity': 1,'minimum': 1,'hintable': True}]))
+                        gbk.minimum_goals = 2
+                    gbk.goal_count = len(gbk.goals)
+                    if (self.settings.lacs_tokens > 0 and self.settings.lacs_condition == 'tokens' and (self.settings.bridge != 'tokens' or self.settings.bridge_tokens < self.settings.lacs_tokens)):
+                        gbk.goals.append(Goal(self, 'Skulls','Light Blue',items=[{'name': 'Gold Skulltula Token','quantity': 100,'minimum': self.lacs_tokens,'hintable': False}]))
+                        gbk.goal_count = round(self.settings.lacs_tokens / 10)
+                        gbk.minimum_goals = 1
+                    if len(gbk.goals) > 0:
+                        self.goal_categories[gbk.name] = gbk
+                elif self.settings.shuffle_ganon_bosskey != 'remove' and self.settings.shuffle_ganon_bosskey != 'vanilla':
+                    gbk = GoalCategory('ganon_bosskey', 30, goal_count=1)
+                    if self.settings.item_pool_value == 'plentiful':
+                        keys = 2
+                    else:
+                        keys = 1
+                    gbk.goals.append(Goal(self, 'the Key','Light Blue',items=[{'name': 'Boss Key (Ganons Castle)','quantity': keys,'minimum': 1,'hintable': True}]))
+                    gbk.minimum_goals = 1
+                    self.goal_categories[gbk.name] = gbk
+
+                if self.skipped_trials['Forest'] == False:
+                    trial_goal.items.append({'name': 'Forest Trial Clear','quantity': 1,'minimum': 1,'hintable': True})
+                    trials.goal_count += 1
+                if self.skipped_trials['Fire'] == False:
+                    trial_goal.items.append({'name': 'Fire Trial Clear','quantity': 1,'minimum': 1,'hintable': True})
+                    trials.goal_count += 1
+                if self.skipped_trials['Water'] == False:
+                    trial_goal.items.append({'name': 'Water Trial Clear','quantity': 1,'minimum': 1,'hintable': True})
+                    trials.goal_count += 1
+                if self.skipped_trials['Shadow'] == False:
+                    trial_goal.items.append({'name': 'Shadow Trial Clear','quantity': 1,'minimum': 1,'hintable': True})
+                    trials.goal_count += 1
+                if self.skipped_trials['Spirit'] == False:
+                    trial_goal.items.append({'name': 'Spirit Trial Clear','quantity': 1,'minimum': 1,'hintable': True})
+                    trials.goal_count += 1
+                if self.skipped_trials['Light'] == False:
+                    trial_goal.items.append({'name': 'Light Trial Clear','quantity': 1,'minimum': 1,'hintable': True})
+                    trials.goal_count += 1
+
+                if int(self.settings.trials) > 0:
```

Comment:
...does this need to be cast to an int? D: Maybe we should make sure to save it as an int, if we're always casting it anyway.

Reply:
I vaguely remember running into type errors at one point, probably here. I can check how it looks without casting.

Reply:
Tested and casting is not necessary with the GUI. I think I was having problems with plando, which #1375 would fix.